### PR TITLE
Fix "Fine-Tune a Semantic Segmentation Model with a Custom Dataset" blog post

### LINKED ISSUE #914: update blog/fine-tune-segformer.md code and inference example
### A few issues:
1. label2id variable starts a { but does not close it.
2. reduce_labels=feature_extractor.reduce_labels,
problem is feature_extractor.reduce_labels does not exist, unless you remove the s at the end... it's feature_extractor.reduce_label

### A suggestion:
At the end of the inference part, to write an example on how to run a image file trough the model and how it can be outputed as a image with the mask applied... or just as a mask (example with code and if it makes sense, explained just like the rest of the article)

### PR DESCRIPTION
**Pull request by segments-tobias**
Closes #914:

- Adds missing bracket to `label2id` code
- Changes `feature_extractor.reduce_labels` to `feature_extractor.do_reduce_labels`, see [https://github.com/huggingface/transformers/pull/21218](https://github.com/huggingface/transformers/pull/21218)

Also adds the notebook to `/notebooks`, and added the notebook gif to `/assets`.

### CODE REVIEW

File: fine-tune-segformer.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ authors:
```
Comment (osanseviero) [inline fine-tune-segformer.md:14]: Could you add the `gif` to https://huggingface.co/datasets/huggingface/documentation-images/tree/main/blog please? That way the GitHub repo remains light and we avoid having too many assets

**Review by osanseviero — APPROVED** [review]
Looks great! Before merging, I have a small suggestion

File: fine-tune-segformer.md (hunk 1 above)
Comment (segments-tobias) [reply fine-tune-segformer.md:14]: Forgot about that 😅 
Here's the PR in the HF dataset with all the assets: [https://huggingface.co/datasets/huggingface/documentation-images/discussions/69](https://huggingface.co/datasets/huggingface/documentation-images/discussions/69).
I left the thumbnail for this blog post in the assets of this repo though.

Comment (osanseviero) [reply fine-tune-segformer.md:14]: Awesome, great! I left a comment on the PR.

Thanks a lot! :fire:

**Comment by segments-tobias** [comment]
Can this be merged? @osanseviero

**Comment by osanseviero** [comment]
Thanks a lot for the fixes! 🔥

**Pull request merged**